// dice-dao/index.js
var AWS = require('aws-sdk');
var dynamo = new AWS.DynamoDB.DocumentClient({
    region: 'ap-northeast-1',
});


module.exports = {
    putDiceUser: function (u) {
        var params = {
            TableName: "dice",
            Item: {
                uuid: u,
                create_time: (new Date()).getTime(),
                dice: 0,
                pos: 99
            },
        };
        return dynamo.put(params).promise();
    },
    getDiceUser: (u) => {
        var params = {
            TableName: "dice",
            Key: {
                uuid: u
            }
        };
        return dynamo.get(params).promise();
    },
    updateDice: (u, dice) => {
        var params = {
            TableName: "dice",
            Key: {
                uuid: u
            },
            UpdateExpression: "set dice = :d",
            ExpressionAttributeValues: {
                ":d": dice
            },
            ReturnValues: "UPDATED_NEW"
        };
        return dynamo.update(params).promise();
    },
    updatePos: (u, pos) => {
        var params = {
            TableName: "dice",
            Key: {
                uuid: u
            },
            UpdateExpression: "set dice = :d, pos = :p",
            ExpressionAttributeValues: {
                ":d": 0,
                ":p": pos
            },
            ReturnValues: "UPDATED_NEW"
        };
        return dynamo.update(params).promise();
    },
    updateName: (u, name) => {
        var params = {
            TableName: "dice",
            Key: {
                uuid: u
            },
            UpdateExpression: "set user_name = :d",
            ExpressionAttributeValues: {
                ":d": name
            },
            ReturnValues: "UPDATED_NEW"
        };
        return dynamo.update(params).promise();
    },
};
